packages/expo-constants/android/src/main/java/expo/modules/constants/ConstantsService.kt

```diff
@@ -54,6 +54,11 @@ open class ConstantsService(private val context: Context) : InternalModule, Cons
 
   override fun getStatusBarHeight() = statusBarHeightInternal
 
+  override fun getIsDevice(): Boolean {
+    // Kept for compatibility with older expo-modules-core. TODO: remove in SDK 51
```

Comment:
I think main is already 51, correct? Curious why we don't want to remove it.

Reply:
```suggestion
    // Kept for backwards compatibility with the interface from older expo-modules-core (SDK 50).
    // TODO: Can be removed once SDK 51 is out.
```

Reply:
Isn't SDK 50 only in the release branch? Should this be a release-branch-only commit?

Reply:
Ah, I was under the impression that we would no longer publish packages from main except with the canary flag. I think the tradeoff of not being able to move the codebase forward in order to retain backwards compatibility has a higher cost than benefit, but it might differ from package to package and you know best about `expo-modules-core`.

In what cases do you anticipate publishing `expo-modules-core` from main going forward? (other than canaries). And if we do publish from main, why wouldn't we do so from the SDK 49 branch (or a branch off of the SDK 49 branch) to have a better record of package compatibility?

Reply:
> we would no longer publish packages from main except with the canary flag

Nope, the canaries should be like nightlies or even used for testing before merging to `main`. It still needs to be possible to publish non-prerelease versions from `main`, from where we usually release new features that can be used with the already released SDKs, but are not part of any SDK yet. In other words, they are not and will not be included in Expo Go until the next SDK release.
We definitely need a faster iteration cycle than SDK releases to get the feedback earlier, make more announcements and especially to be nice for contributors that took their time to implement features they really need in their app without forcing them to wait months for the next SDK.

> In what cases do you anticipate publishing `expo-modules-core` from main going forward?

This package should be backwards compatible as much as possible, because it's also used by library authors and we don't want them to enforce their users to use only the specific SDK. Thus I don't see any case when we need to release `expo-modules-core` from main. Another reason for this is that it's only a dependency of the `expo` package which is released only from the SDK branch 😉 

> And if we do publish from main, why wouldn't we do so from the SDK 49 branch (or a branch off of the SDK 49 branch) to have a better record of package compatibility?

Probably we could do it from the SDK branch, but it requires more effort - switching to another diverged branch, checking if it's even possible to cherry-pick these changes, another round of testing, syncing changelogs, and so on. Moreover, we usually don't want such changes to be included in Expo Go.

Reply:
I see. I worry that doing these releases from `main` leaves us in a state where only a select few people on the team know what the current state of the releases are. This makes things like debugging/triaging harder as we now have more combinations of versions that we support and thus more surface area. I guess my main concern is that there's just a lot of surface area as it is, and supporting cross-versioning adds attention cost both for support but also for development on main since we now need to be aware of all supported cross-versions when removing/deprecating/adding code. The mental model I find simplest is that all packages are normally published in the same way (from SDK branches) and canaries are published from main. That way, we can confidently say that if a person is on SDK X then the versions of their packages have been QA'd during the QA cycle and also are limited to those on the SDK branch (rather than needing to check out multiple versions of the repository to debug packages from multiple branches).

> We definitely need a faster iteration cycle than SDK releases to get the feedback earlier, make more announcements and especially to be nice for contributors that took their time to implement features they really need in their app without forcing them to wait months for the next SDK.

I think the correct solution here is to just do more frequent SDK releases. Thankfully we're moving towards being able to do that.